File: android/app/src/main/java/com/lightningkite/kwifttemplate/shared/api/APIOnline.kt
///Kotlin Only
@file:Suppress("ConvertToStringTemplate", "RemoveExplicitTypeArguments")

///End Kotlin Only

package com.lightningkite.kwifttemplate.shared.api

import com.lightningkite.kwift.actual.HttpClient
import com.lightningkite.kwift.actual.escaping
import com.lightningkite.kwifttemplate.shared.models.Post


class APIOnline(val baseUrl: String = "https://jsonplaceholder.typicode.com") : APIInterface {
    override fun getExamplePosts(onResult: @escaping() (Int, List<Post>?, String?) -> Unit) {
        HttpClient.call(
            url = baseUrl + "/posts",
            method = HttpClient.GET,
            headers = mapOf(),
            body = null,
            onResult = onResult
        )
    }

}

File: ios/project/shared/api/APIOnline.swift
import Foundation
///Kotlin Only
//@file:Suppress("ConvertToStringTemplate", "RemoveExplicitTypeArguments")
//
///End Kotlin Only

//package com.lightningkite.kwifttemplate.shared.api


//import com.lightningkite.kwift.actuals.*
//import com.lightningkite.kwifttemplate.shared.models.*
//import java.util.*


final public class APIOnline : APIInterface {
    
    public var baseUrl: String 
    
    init( baseUrl: String = "https://jsonplaceholder.typicode.com") {
        self.baseUrl = baseUrl
    }
    convenience init(
        _ baseUrl: String 
    ){ 
        self.init(
            baseUrl: baseUrl
        ) 
    }
    
    public func getExamplePosts( onResult: @escaping (Int, [Post]?, String?) -> Void) {
        HttpClient.call(
            url : baseUrl + "/posts",
            method : HttpClient.GET,
            headers : [String: String](),
            body : nil,
            onResult : onResult
        )
    }
    
}
